json/json_test.go

```diff
@@ -93,3 +103,31 @@ func TestService(t *testing.T) {
 		t.Errorf("Expected http response code 400, but got %v", code)
 	}
 }
+
+func TestServiceBeforeAfter(t *testing.T) {
+	s := rpc.NewServer()
+	s.RegisterCodec(NewCodec(), "application/json")
+	service := Service1{}
+	service.beforeAfterContext = map[string]string{}
+	s.RegisterService(new(service), "")
```

Comment:
Syntax error is causing tests to fail here I believe you want to get rid of the `new`